src/main/java/org/zalando/nakadi/controller/CursorOperationsController.java

```diff
@@ -0,0 +1,196 @@
+        accessControl(eventTypeName, client);
+
+        final List<ShiftedNakadiCursor> domainCursor = cursors.getList().stream()
+                .map(this.toShiftedNakadiCursor(eventTypeName))
+                .collect(Collectors.toList());
+
+        final List<NakadiCursor> domainResultCursors = cursorOperationsService.unshiftCursors(domainCursor);
+
+        final List<Cursor> viewResult = domainResultCursors.stream().map(cursorConverter::convert)
+                .collect(Collectors.toList());
+
+        return status(OK).body(viewResult);
+    }
+
+    @RequestMapping(path = "/event-types/{eventTypeName}/cursors-lag", method = RequestMethod.POST)
+    public ResponseEntity<?> cursorsLag(@PathVariable("eventTypeName") final String eventTypeName,
+                                        @Valid @RequestBody final ValidListWrapper<Cursor> cursors,
+                                        final Client client) {
+        accessControl(eventTypeName, client);
+
+        final List<NakadiCursor> domainCursor = cursors.getList().stream()
+                .map(toNakadiCursor(eventTypeName))
+                .collect(Collectors.toList());
+
+        final List<NakadiCursorLag> lagResult = cursorOperationsService
+                .cursorsLag(eventTypeName, domainCursor);
+
+        final List<CursorLag> viewResult = lagResult.stream().map(this::toCursorLag)
+                .collect(Collectors.toList());
+
+        return status(OK).body(viewResult);
+    }
+
+    @ExceptionHandler(InvalidCursorOperation.class)
+    public ResponseEntity<?> invalidCursorOperation(final InvalidCursorOperation e,
+                                                          final NativeWebRequest request) {
+        LOG.debug("User provided invalid cursor for operation. Reason: " + e.getReason(), e);
+        return Responses.create(Problem.valueOf(MoreStatus.UNPROCESSABLE_ENTITY,
+                clientErrorMessage(e.getReason())), request);
+    }
+
+    private String clientErrorMessage(final InvalidCursorOperation.Reason reason) {
+        switch (reason) {
+            case INVERTED_TIMELINE_ORDER: return "Inverted timelines. Final cursor must correspond to a newer " +
+                    "timeline than initial cursor.";
+
+            case TIMELINE_NOT_FOUND: return "Timeline not found. It might happen in case the cursor refers to a " +
+                    "timeline that has already expired.";
+            case INVERTED_OFFSET_ORDER: return "Inverted offsets. Final cursor offsets must be newer than initial " +
+                    "cursor offsets";
+            case PARTITION_NOT_FOUND: return "Partition not found.";
+            case CURSORS_WITH_DIFFERENT_PARTITION: return "Cursors with different partition. Pairs of cursors should " +
+                    "have matching partitions.";
+            default: return unexpectedErrorReason(reason);
+        }
+    }
+
+    private void accessControl(final String eventTypeName, final Client client) {
```

Comment:
Maybe it's better to select different name (one of verbs because noun is not really describing what is going to happen), I think that `checkReadScopes` or `verifyCanRead` matches better.